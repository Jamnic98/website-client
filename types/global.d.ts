type navlinkObjType = { label: string; url: string; children: navlinkObj[] };

type projectObjType = {
  id: string;
  title: string;
  shortDescription: string;
  longDescription: string[];
  projectPageURI: string;
  screenshotURIs: string[];
  mainLanguage: string;
  links: projectLinkObjType[];
  techStack: string[];
};

type projectLinkObjType = { URL: string; label: string; type: string };

type headerObjType = {
  title: string;
  description: string | null;
};

type runDataType = {
  distance: number;
  duration: number;
  start_date_local: date;
};

export { navlinkObjType, projectObjType, projectLinkObjType, headerObjType, runDataType };
