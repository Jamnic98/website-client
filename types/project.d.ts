export type ProjectLinkData = { URL: string; label: string; type: string };

export type ProjectData = {
  id: string;
  title: string;
  summary: string;
  paragraphs: string[];
  projectPageURI: string;
  screenshotURIs: string[];
  mainLanguage: string;
  links: ProjectLinkData[];
  techStack: string[];
};
