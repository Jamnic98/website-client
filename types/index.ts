export * from './navlink.types'
export * from './project.types'
export * from './runs.types'
