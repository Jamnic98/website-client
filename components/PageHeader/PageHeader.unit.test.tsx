import Header from './PageHeader';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { headerObj } from '../../types/global';

const testHeaderObj: headerObj = {
  title: 'Title',
  description: 'This is a test.'
};

describe('Header', () => {
  it('renders correctly', () => {
    const { title, description } = testHeaderObj;
    render(<Header headerObj={testHeaderObj} />);
    const header = screen.getByTestId('header');
    expect(header).toHaveTextContent(`${title}${description}`);
  });
  
  
  it('renders Header unchanged', () => {
    const tree = renderer
      .create(
        <Header headerObj={testHeaderObj} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
