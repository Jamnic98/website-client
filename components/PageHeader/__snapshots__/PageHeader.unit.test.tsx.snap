// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header unchanged 1`] = `
<header
  className="header"
  data-testid="header"
>
  <h1
    className="title"
  >
    Title
  </h1>
  <p
    className="description"
  >
    This is a test.
  </p>
  <hr
    className="hr"
  />
</header>
`;
